WordPress/gutenberg
# Interactivity API: migration to the new `store()` API

## What?

This PR migrates the runtime and core blocks to the new `store()` API, described in:

- https://github.com/WordPress/gutenberg/discussions/53586

Part of https://github.com/WordPress/gutenberg/issues/53740.

## Why?

The benefits of the new `store()` API are outlined in this discussion:

- https://github.com/WordPress/gutenberg/discussions/53586

## How?

- Coping/pasting the Interactivity API runtime code from this PR:
   https://github.com/woocommerce/woocommerce-blocks/pull/11071
- Updating all the e2e tests so they use the new API.
- Migrating all the Core blocks.

## Testing Instructions

- Add the different interactive Core blocks (Navigation, Image, File, Search and Query).
- Make sure that they keep working exactly as in `trunk`.
